src/main/java/org/apache/cassandra/sidecar/data/StreamSSTableComponentRequest.java

```diff
@@ -112,6 +127,39 @@ public String toString()
                ", snapshot='" + snapshotName + '\'' +
                ", secondaryIndexName='" + secondaryIndexName() + '\'' +
                ", componentName='" + componentName() + '\'' +
+               ", dataDirectoryIndex='" + dataDirectoryIndex + '\'' +
                '}';
     }
+
+    public static StreamSSTableComponentRequest from(QualifiedTableName qualifiedTableName, RoutingContext context)
+    {
+        String snapshotName = context.pathParam("snapshot");
+        String secondaryIndexName = context.pathParam("index");
+        String componentName = context.pathParam("component");
+        String tableId = maybeGetTableId(context.pathParam("table"));
+        Integer dataDirectoryIndex = RequestUtils.parseIntegerQueryParam(context.request(), "dataDirectoryIndex", null);
```

Comment:
By assigning the default value 0 here, the implement can be simply by avoiding the null case. 

```suggestion
        int dataDirectoryIndex = RequestUtils.parseIntegerQueryParam(context.request(), "dataDirectoryIndex", 0);
```